staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -1338,3 +1374,48 @@ func buildOpenAPIModelsForApply(staticOpenAPISpec *goopenapispec.Swagger, crd *a
 	}
 	return models, nil
 }
+
+func (r *crdHandler) updateStorageVersionFor(crd *apiextensionsv1.CustomResourceDefinition) error {
+	gr := schema.GroupResource{
+		Group:    crd.Spec.Group,
+		Resource: crd.Spec.Names.Plural,
+	}
+	storageVersion, err := apiextensionshelpers.GetCRDStorageVersion(crd)
+	if err != nil {
+		// This should never happened if crd is valid, which is true since we
+		// only update storage version for CRDs that have been written to the
+		// storage
+		return err
+	}
+	encodingVersion := crd.Spec.Group + "/" + storageVersion
+	var decodableVersions []string
+	for _, v := range crd.Spec.Versions {
+		if !v.Served {
```

Comment:
The doc says "Served is a flag enabling/disabling this version from being served via REST APIs", even if a version is not served, it's still decodable, right?

Reply:
it is